polkadot/runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -977,18 +1077,16 @@ fn sanitize_backed_candidates<T: crate::inclusion::Config>(
 	// Map the candidates to the right paraids, while making sure that the order between candidates
 	// of the same para is preserved.
 	let mut candidates_per_para: BTreeMap<ParaId, Vec<_>> = BTreeMap::new();
+
+	// Get the paras scheduled next on each core.
+	let scheduled_paras = scheduler::Pallet::<T>::scheduled_paras().collect::<BTreeMap<_, _>>();
 	for candidate in backed_candidates {
-		// Drop any v2 candidate receipts if nodes are not allowed to use them.
-		// It is mandatory to filter these before calling `filter_unchained_candidates` to ensure
-		// any v1 descendants of v2 candidates are dropped.
-		if !allow_v2_receipts && candidate.descriptor().version() == CandidateDescriptorVersion::V2
-		{
-			log::debug!(
-				target: LOG_TARGET,
-				"V2 candidate descriptors not allowed. Dropping candidate {:?} for paraid {:?}.",
-				candidate.candidate().hash(),
-				candidate.descriptor().para_id()
-			);
+		if !sanitize_backed_candidate_v2::<T>(
+			&candidate,
+			allowed_relay_parents,
+			&scheduled_paras,
```

Comment:
here we only really care to find out the cores that this para is scheduled on (which we already have as part of the `scheduled: BTreeMap<ParaId, BTreeSet<CoreIndex>>` parameter on this function).
No need to query the scheduled paras and build another collection

Reply:
🤦🏼‍♂️ didn't notice it.

Reply:
Actually this one is different, it maps cores to the claims at cq offset: `BTreeMap<CoreIndex, ParaId>`

Reply:
fixed